ingestion/src/main/java/feast/store/serving/redis/FeatureRowToRedisMutationDoFn.java

```diff
@@ -64,14 +64,20 @@ private RedisKey getKey(FeatureRow featureRow) {
     return redisKeyBuilder.build();
   }
 
+  private byte[] getValue(FeatureRow featureRow) {
+    FeatureRowEncoder encoder =
+        new FeatureRowEncoder(featureSets.get(featureRow.getFeatureSet()).getSpec());
+    return encoder.encode(featureRow).toByteArray();
+  }
+
   /** Output a redis mutation object for every feature in the feature row. */
   @ProcessElement
   public void processElement(ProcessContext context) {
     FeatureRow featureRow = context.element();
     try {
-      RedisKey key = getKey(featureRow);
-      RedisMutation redisMutation =
-          new RedisMutation(Method.SET, key.toByteArray(), featureRow.toByteArray(), null, null);
+      byte[] key = getKey(featureRow).toByteArray();
+      byte[] value = getValue(featureRow);
```

Comment:
Since this is a Redis specific implementation, wouldn't it make more sense for the `encoder` to be a part of the value builder (`getValue`)? I don't really see a good reason to externalize it into its own class.

So basically the logic from `encode` can all be in `getValue`

Reply:
Externalize into it's own class is mainly for the ease of testing, unless we are going to expose getValue (package private or public).

Reply:
It should be pretty similar to test right? In fact it might even be better to test the actual byte values that get written to Redis because we have had a lot of problems there in the past.

Reply:
Another approach is changing the encoder class to `RedisValueEncoder` or something so that it actually builds the final byte values as well.

Reply:
I have made the encoding process part of the DoFn and change the test accordingly.